test/common/dlock_test.cpp

```diff
@@ -0,0 +1,174 @@
+/*
+ *  Copyright (c) 2020 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Project: curve
+ * Created Date: April 2nd 2021
+ * Author: wanghai01
+ */
+
+#include <gtest/gtest.h>
+#include <glog/logging.h>
+#include <thread>
+#include <chrono>
+#include <cstdlib>
+#include <memory>
+#include <string>
+#include <ctime>
+#include "src/common/timeutility.h"
+#include "src/kvstorageclient/etcd_client.h"
+#include "src/common/concurrent/dlock.h"
+
+namespace curve {
+namespace common {
+
+using ::curve::kvstorage::EtcdClientImp;
+
+class TestDLock : public ::testing::Test {
+ protected:
+    TestDLock() {}
+    ~TestDLock() {}
+
+    void SetUp() override {
+        system("rm -fr testDLock.etcd");
+        client_ = std::make_shared<EtcdClientImp>();
+        char endpoints[] = "127.0.0.1:2375";
```

Comment:
testport record in curve/test/resources.list

Reply:
done